coordinator/internal/authority/authority.go

```diff
@@ -68,33 +81,57 @@ func New(hist *history.History, reg *prometheus.Registry, log *slog.Logger) *Aut
 	})
 	manifestGeneration.Set(0)
 
-	return &Authority{
-		bundles: make(map[string]Bundle),
-		hist:    hist,
-		logger:  log.WithGroup("mesh-authority"),
+	ticker := clock.RealClock{}.NewTicker(24 * time.Hour)
+
+	a := &Authority{
+		bundles:   make(map[string]Bundle),
+		hist:      hist,
+		logger:    log.WithGroup("mesh-authority"),
+		kdsGetter: snp.NewCachedHTTPSGetter(memstore.New[string, []byte](), ticker, logger.NewNamed(log, "kds-getter")),
+		attestationFailuresCounter: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Subsystem: "contrast_meshapi",
+			Name:      "attestation_failures_total",
+			Help:      "Number of attestation failures from workloads to the Coordinator.",
+		}),
 		metrics: metrics{
 			manifestGeneration: manifestGeneration,
 		},
 	}
+
+	a.atlsCredentials.Store(atlscredentials.New(atls.NoIssuer, []atls.Validator{}))
+
+	return a
 }
 
 // SNPValidateOpts returns SNP validation options from reference values.
 //
 // It also ensures that the policy hash in the report's HOSTDATA is allowed by the current
 // manifest.
-func (m *Authority) SNPValidateOpts(report *sevsnp.Report) (*validate.Options, error) {
+func (m *Authority) SNPValidateOpts() ([]func(report *sevsnp.Report) (*validate.Options, error), error) {
 	state := m.state.Load()
 	if state == nil {
 		return nil, ErrNoManifest
 	}
 	mnfst := state.manifest
 
-	hostData := manifest.NewHexString(report.HostData)
-	if _, ok := mnfst.Policies[hostData]; !ok {
-		return nil, fmt.Errorf("hostdata %s not found in manifest", hostData)
+	validateOpts, err := mnfst.AKSValidateOpts()
+	if err != nil {
+		return nil, fmt.Errorf("get validate options: %w", err)
 	}
 
-	return mnfst.AKSValidateOpts()
```

Comment:
This would have been the straight-forward way to implement it, which I also considered when writing this. However, it seemed tricky to do, since we wrap the additional `hostData` check in the actual options generator. Do we even require this check at this point, or could we just *always* pass `hostData`?

Reply:
Conclusion: Possible. Move logic out of the authority and into the Manifest.

Reply:
Reiterating on this, I think what we could do is implement some sort of combinator in the manifest package that sugars each of the `[]*validate.Options` with exactly that `hostData` check. Is that what you referred to initially @burgerdev?